crates/sdk/src/tx.rs

```diff
@@ -3930,52 +3906,53 @@ async fn target_exists_or_err(
 }
 
 /// Returns the given refund target address if the given address is valid for
-/// the IBC shielded transfer. Returns an error if the address is transparent
-/// or the address is given for non-shielded transfer.
+/// the IBC shielded transfer. Returns an error if the address is a payment
+/// address or given for non-shielded transfer.
 async fn get_refund_target(
     context: &impl Namada,
     source: &TransferSource,
     refund_target: &Option<TransferTarget>,
-) -> Result<Option<PaymentAddress>> {
+) -> Result<Option<Address>> {
     match (source, refund_target) {
-        (
-            TransferSource::ExtendedSpendingKey(_),
-            Some(TransferTarget::PaymentAddress(pa)),
-        ) => Ok(Some(*pa)),
+        (_, Some(TransferTarget::PaymentAddress(pa))) => {
+            Err(Error::Other(format!(
+                "Supporting only a transparent address as a refund target: {}",
+                pa,
+            )))
+        }
         (
             TransferSource::ExtendedSpendingKey(_),
             Some(TransferTarget::Address(addr)),
-        ) => Err(Error::Other(format!(
-            "Transparent address can't be specified as a refund target: {}",
-            addr,
-        ))),
-        (TransferSource::ExtendedSpendingKey(spending_key), None) => {
-            // Generate a new payment address
-            let viewing_key =
-                ExtendedFullViewingKey::from(&(*spending_key).into()).fvk.vk;
+        ) => Ok(Some(addr.clone())),
+        (TransferSource::ExtendedSpendingKey(_), None) => {
+            // Generate a new transparent address if it doesn't exist
             let mut rng = OsRng;
-            let (div, _g_d) = crate::masp::find_valid_diversifier(&mut rng);
-            let payment_addr: PaymentAddress = viewing_key
-                .to_payment_address(div)
-                .ok_or_else(|| {
-                    Error::Other(
-                        "Converting to a payment address failed".to_string(),
-                    )
-                })?
-                .into();
-            let alias = format!("ibc-refund-target-{}", rng.next_u64());
             let mut wallet = context.wallet_mut().await;
+            if let Some(addr) = wallet.find_address(IBC_REFUND_TARGET_ALIAS) {
```

Comment:
I believe it's better to generate a new disposable address every time, otherwise we might establish some linkage between the shielded ibc operations and this transparent refund address

Reply:
That's a default one. A user can prepare an address with `--refund-target`.

Reply:
But yes, it's better to generate a new one for useability